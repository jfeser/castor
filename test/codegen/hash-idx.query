AHashIdx(Dedup(Select([f], r1)) as k, AList(Filter(f = k.f, r1) as lk, ascalar(lk.g)), 2)