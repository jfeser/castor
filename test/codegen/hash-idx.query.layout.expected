0:1 Table len (=20)
1:1 Table map len (=4)
1:0 Table hash
1:0 Table hash len (=0)
2:4 Table key map
2:1 Map entry (0 => 0)
3:1 Map entry (1 => 0)
4:1 Map entry (2 => 5)
5:1 Map entry (3 => 10)
6:14 Table values
6:1 Scalar (=(Int 1))
7:1 List count (=2)
8:1 List len (=4)
9:2 List body
9:1 Scalar (=(Int 2))
10:1 Scalar (=(Int 3))
11:1 Scalar (=(Int 2))
12:1 List count (=2)
13:1 List len (=4)
14:2 List body
14:1 Scalar (=(Int 1))
15:1 Scalar (=(Int 2))
16:1 Scalar (=(Int 3))
17:1 List count (=1)
18:1 List len (=3)
19:1 List body
19:1 Scalar (=(Int 4))
